packages/react/src/utils/useIntersectionRef.ts

```diff
@@ -12,7 +12,7 @@ import { useRef, useEffect } from 'react';
  * return <span ref={elementRef}>...</span>
  */
 export default function useIntersectionRef<T extends HTMLElement>(
-  element: T | MutableRefObject<T>,
```

Comment:
All of our behavior hooks follow a similar pattern of where they:

- Accept an element
- Or accept a ref

See:
- https://github.com/dequelabs/cauldron/blob/develop/packages/react/src/utils/useFocusTrap.ts
- https://github.com/dequelabs/cauldron/blob/develop/packages/react/src/utils/useEscapeKey.ts
- https://github.com/dequelabs/cauldron/blob/develop/packages/react/src/utils/resolveElement.ts

(Note: there's some additional work we need to do to actually align on the interface, but the underlying principal is there; and unfortunately not well documented right now)

I'm not sure we have a convincing argument yet for removing the element interface to this hook.